test/functional/page_objects/settings_page.js

```diff
@@ -45,7 +46,11 @@ export function SettingsPageProvider({ getService, getPageObjects }) {
 
     async clickKibanaSettings() {
       await this.clickLinkText('Advanced Settings');
-      await PageObjects.header.waitUntilLoadingHasFinished();
+      const isSettingsLoaded = await testSubjects.exists('managementSettingsTitle');
+
+      // Verify navigation is successful, or else fail the test consuming this.
+      expect(isSettingsLoaded).to.be(true);
+      return isSettingsLoaded;
```

Comment:
Why return the value here? Looks like it's not used anywhere, and the name of the function doesn't really indicate it.

Reply:
An `async` method expects a promise as a return value or if a promise isn't returned then it wraps the return value in `Promise.resolve`. I don't really know whether that means it's OK to return undefined or not, but resolving undefined feels unexpected to me. I don't have a strong feeling on this so if you feel like something else would make more sense, I'll roll with that.

Reply:
Resolve to undefined makes more sense to me then resolving to true, as it would take looking into the function to understand what the value of true indicated. It doesn't even really mean anything because it always returns true, or fails. When I think of a function resolving to `true`, I think it can also resolve to `false`.  Not that the option is `resolve(true)` or `reject()`.  Anyway, most `clickX` functions don't return anything, so for the sake of consistency, I think we should follow suit.

Reply:
Works for me, I'll change it. Thanks!